cache.go

```diff
@@ -318,8 +432,213 @@ func (c *GoBuildCache) ToRunOption(distroKey string, opts ...GoBuildCacheOption)
 	})
 }
 
-// BazelCache sets up a cache for bazel builds.
-//
+// CargoBuildCache is a cache for Rust/Cargo build artifacts.
+// It uses sccache to speed up Rust compilation by caching build artifacts.
+type CargoBuildCache struct {
+	// Scope adds extra information to the cache key.
+	// This is useful to differentiate between different build contexts if required.
+	//
+	// This is mainly intended for internal testing purposes.
+	Scope string `json:"scope,omitempty" yaml:"scope,omitempty"`
+
+	// The cargobuild cache may be automatically injected into a build if
+	// rust is detected.
+	// Disabled explicitly turns this off.
+	Disabled bool `json:"disabled,omitempty" yaml:"disabled,omitempty"`
+}
+
+func (c *CargoBuildCache) validate() error {
+	return nil
+}
+
+type CargoBuildCacheInfo struct {
+	Platform *ocispecs.Platform
+}
+
+type CargoBuildCacheOption interface {
+	SetCargoBuildCacheOption(*CargoBuildCacheInfo)
+}
+
+type CargoBuildCacheOptionFunc func(*CargoBuildCacheInfo)
+
+func (f CargoBuildCacheOptionFunc) SetCargoBuildCacheOption(info *CargoBuildCacheInfo) {
+	f(info)
+}
+
+func WithCargoCacheConstraints(opts ...llb.ConstraintsOpt) CacheConfigOption {
+	return CacheConfigOptionFunc(func(info *CacheInfo) {
+		var c llb.Constraints
+		for _, opt := range opts {
+			opt.SetConstraintsOption(&c)
+		}
+		info.CargoBuild.Platform = c.Platform
+	})
+}
+
+const (
+	sccacheCacheDir    = "/tmp/dalec/sccache-cache"
+	sccacheBinary      = "/tmp/dalec/sccache"
+	sccacheCacheDirWin = "C:\\temp\\dalec\\sccache-cache"
+	sccacheBinaryWin   = "C:\\temp\\dalec\\sccache.exe"
+)
+
+func (c *CargoBuildCache) ToRunOption(worker llb.State, distroKey string, opts ...CargoBuildCacheOption) llb.RunOption {
```

Comment:
I realize we probably don't need to pass a worker in here since we can grab `ei.State`.